### DISCUSSION TITLE
Projects

### DISCUSSION THREAD

**Discussion by bryanrcarlson** [CafIncubator/Midden#93] [Q&A]
Hi All -- I'm working on implementing Projects and want to get some feedback on the different options of implementation.

Currently, a project file is implemented by using the file extension ".mippen" for a text/markdown file. But, as we discussed, using an unknown/rare file extension like this makes it difficult for people to open the file if they are browsing the directory structure. It also removes any advantage that a markdown editor might provide as the editor would not recognize the file extension.

We discussed a better alternative might be to rename the file with the markdown extension (.md) and have YAML front-matter to specify that the markdown file is a project file. It was assumed that this markdown file will still have the convention of being named the same as the project it is defining (e.g. "MyAwesomeProject.md").

I'm considering a third option now. I think instead of naming the file with the project name, we could use a consistent filename to indicate it is a project description and use front-matter to define what project the file is describing. I think the filename could also be something more user-friendly -- something that a user browsing the file structure might open if they are trying to find more information about the files. Something like "DESCRIPTION.md" or "PROJECT.md".

So:

- Option 1: "MyAwesomeProject.mippen"
- Option 2: "MyAwesomeProject.md"
- Option 3: "DESCRIPTION.md"

What would you prefer?

**Comment by emilyernst** [comment]
My rankings: option 3 > option 2 > option 1

**Comment by riemino** [comment]
My vote would go for a combination between 2 and 3. Something like, KendallMet_MPROJECT.md
In the case where you might want to have two project files in the same directory. Not sure how common that would be. Also, by having and M as a prefix to PROJECT, this could allow us to identify it as a Midden project.
Just my 2 centavos. 😉

**Reply by bryanrcarlson** [reply]
Oh yeah, good point on multiple files in the same directory. I really like the cleanness and simplicity of something like "DESCRIPTION" though. It's also descriptive of what it is/does; which will steer a "clueless" collaborator towards opening the file as they browse.

How about KendallMet_DESCRIPTION.md? Or, maybe, DESCRIPTION_KendallMet.md?

**Reply by bryanrcarlson** [reply]
Counterpoint: having the convention of project description files having the same name (e.g. "DESCRIPTION.md") will enforce best practices of having a directory for each project. This will also subtly push towards grouping datasets with project directories (and descriptions) instead of having a single directory as a "dumping ground" for project descriptions.

**Reply by emilyernst** [reply]
I agree with you Bryan - you can always have the project name in within the markdown file, too.
